[DOCS] Creates one pager for outlier detection

szabosteve opened the pull request:
## Overview

This PR:
* creates `Finding outliers` page
* moves`Outlier detection`, `Outlier detection evaluation`, and `Finding outliers in eCommerce sample data` sections under `Finding outliers`
* adds metadata to `Finding outliers` page
* adds `ml-dfa-shared.asciidoc` file

Related to https://github.com/elastic/stack-docs/issues/1725

## Note

**Do not merge before https://github.com/elastic/elasticsearch/pull/74745 and https://github.com/elastic/stack-docs/pull/1733**

### Preview

[Book TOC](https://stack-docs_1732.docs-preview.app.elstc.co/guide/en/machine-learning/master/index.html)

szabosteve commented:
@elasticmachine test this please

lcawl commented on docs/en/stack/ml/df-analytics/ml-dfa-outlier-detection.asciidoc:
```diff
@@ -1,9 +1,201 @@
+:description: An introduction to {ml} {oldetection}, which enables you to  \
+find unusual data points in a data set compared to the normal data points.
+
+This page contains a conceptual ovierview of 
```
```suggestion
This page contains a conceptual overview of 
```

lcawl commented on docs/en/stack/ml/df-analytics/ml-dfa-outlier-detection.asciidoc:
```diff
@@ -1,9 +1,201 @@
+find unusual data points in a data set compared to the normal data points.
+
+This page contains a conceptual ovierview of 
+<<dfa-outlier-detection,{oldetection}>> in the {stack} and a simple
```
IMO since it's targeting the next paragraph, this link is unnecessary
```suggestion
{oldetection} in the {stack} and a simple
```

lcawl commented on docs/en/stack/ml/df-analytics/ml-dfa-outlier-detection.asciidoc:
```diff
@@ -1,9 +1,201 @@
+feature values are different from those of the normal data points in a 
+particular data set. Outliers may denote errors or unusual behavior.
+
+We use unsupervised {oldetection} which means there is no need to provide a 
```
I think in this case it's best to be clear about who/what "we" is:
```suggestion
The {ml-features} provide unsupervised {oldetection}, which means there is no need to provide a 
```

lcawl commented on docs/en/stack/ml/df-analytics/ml-dfa-outlier-detection.asciidoc:
```diff
@@ -1,9 +1,201 @@
+particular data set. Outliers may denote errors or unusual behavior.
+
+We use unsupervised {oldetection} which means there is no need to provide a 
+training data set to teach {oldetection} to recognize outliers. Unsupervised 
```
Minor suggestion, but I think the end of this sentence is unnecessary.
```suggestion
training data set. Unsupervised 
```

lcawl commented:
This is already looking great!  A few things that I think might be nice to add in this or future iterations:

- Before going into the low level details about the formulas, it might be nice to have a few sentences about the types of problems/use cases this analysis solves well.
- Before talking about the types of evaluation, it would be nice to briefly reiterate the steps you have to do before then (e.g. define the problem, curate the data set, create the job, etc)

szabosteve commented:
> * Before going into the low level details about the formulas, it might be nice to have a few sentences about the types of problems/use cases this analysis solves well.
> * Before talking about the types of evaluation, it would be nice to briefly reiterate the steps you have to do before then (e.g. define the problem, curate the data set, create the job, etc)

I addressed these points via 08f0dac.
I summarized the steps before evaluation in the eval intro in the shared file, so it'll present in every type of analysis.
